### PR TITLE
Fixes upgrade from 0.63.x to 1.3.x AC switch OFF and wallet creation failed

### PR DESCRIPTION
**Pull request by tmancey**
Fixes https://github.com/brave/brave-browser/issues/7833

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [x] iOS
  - [x] Linux
  - [x] macOS
  - [x] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [x] iOS
  - [x] Linux
  - [x] macOS
  - [x] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

See https://github.com/brave/brave-browser/issues/7833 (resets Ads viewed count to zero is a known issue and was agreed that migration would not be attempted for https://github.com/brave/brave-browser/issues/4047)

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/state/wallet_state.cc [hunk 1]
```diff
@@ -75,22 +75,17 @@ bool WalletState::FromDict(
...
-  if (!choices_bat_list) {
-    NOTREACHED();
-    return false;
-  }
-
-  for (const auto&choices_bat_value : choices_bat_list->GetList()) {
-    const double bat = choices_bat_value.GetDouble();
-    wallet_properties.parameters_choices.push_back(bat);
+  if (choices_bat_list) {
+    for (const auto&choices_bat_value : choices_bat_list->GetList()) {
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/state/wallet_state.cc]: Really minor, but maybe a space after the `&`?

**Comment by bsclifton** [comment]
CI run was completely green 🎉 I did a quick commit to address that nit, @emerick 😄 Should be ready to merge!

**Comment by tmancey** [comment]
Sanity tests of fix https://github.com/brave/brave-core/pull/4449 have shown that users who experienced this issue will have their data recovered after upgrading as long the user did not try to re-enable Brave Rewards. If a user did re-enable Rewards before upgrading to this fix then the wallet will be corrupted and is non-recoverable.

Sanity testing was from 0.63.55 to 1.3, 1.4 and 1.5, however, we should test other scenarios.

As a side note P3A is broken for wallet creation as it still references Probi etc. which does not exist in the JSON feed. I will raise a separate issue for this.

**Comment by LaurenWags** [comment]
Verified using

Brave | 1.5.69 Chromium: 79.0.3945.130 (Official Build) nightly (64-bit)
-- | --
Revision | e22de67c28798d98833a7137c0e22876237fc40a-refs/branch-heads/3945@{#1047}
OS | macOS Version 10.14.6 (Build 18G103)

**PASS** Upgrading without hitting the issue first:
1. 0.63.55
   - Install above version.
   - Enable Rewards, view an ad
   - Close. Rename profile to Nightly.
   - Launch 1.5.69
   - Ads panel displays as expected, AC is still on as expected, no error on wallet panel as expected

2. 0.64.77
   - Install above version.
   - Enable Rewards, view an ad
   - Close. Rename profile to Nightly.
   - Launch 1.5.69
   - Ads panel displays as expected, AC is still on as expected, no error on wallet panel as expected

**PASS** Upgrading after hitting the issue first:
3. 0.63.55
   - Install above version.
   - Enable Rewards, view an ad
   - Close. Rename profile to Nightly. Download Nightly 1.5.53.
   - Launch 1.5.53 --> see issue but don't do anything.
   - Close. Update to 1.5.69 on Nightly and relaunch.
   - Ads panel displays as expected, AC is still on as expected, no error on wallet panel as expected

4. 0.64.77
   - Install above version.
   - Enable Rewards, view an ad
   - Close. Rename profile to Nightly. Download Nightly 1.5.53.
   - Launch 1.5.53 --> see issue but don't do anything.
   - Close. Update to 1.5.69 on Nightly and relaunch.
   - Ads panel displays as expected, AC is still on as expected, no error on wallet panel as expected

Note - see https://github.com/brave/brave-core/pull/4449#issuecomment-579525451 for additional information

**PASS** Sanity check of other upgrades:
5. 0.69.135
   - Install above version.
   - Enable Rewards, view an ad
   - Close. Rename profile to Nightly.
   - Launch 1.5.69
   - Ads panel displays as expected, AC is still on as expected, no error on wallet panel as expected

6. 1.2.43
   - Install above version.
   - Enable Rewards, view an ad
   - Close. Rename profile to Nightly.
   - Launch 1.5.69
   - Ads panel displays as expected, AC is still on as expected, no error on wallet panel as expected

**Pull request merged**